web/controllers/v1/relay_group_membership_controller.ex

```diff
@@ -12,24 +12,33 @@ defmodule Cog.V1.RelayGroupMembershipController do
     render(conn, Cog.V1.RelayGroupView, "relays.json", relay_group: relay_group)
   end
 
+  def manage_membership(conn, %{"id" => id, "relays" => member_spec}),
+  do: manage_association(conn, :relay, %{"id" => id, "members" => member_spec})
+
+  def manage_assignment(conn, %{"id" => id, "bundles" => member_spec}),
+  do: manage_association(conn, :bundle, %{"id" => id, "members" => member_spec})
+
   # Manage membership of a relay group. Adds and deletes can be submitted and
   # processed in a single request.
-  def manage_membership(conn, %{"id" => id, "relays" => member_spec}) do
+  def manage_association(conn, type, %{"id" => id, "members" => member_spec}) do
     result = Repo.transaction(fn() ->
       group = Repo.get!(RelayGroup, id)
 
-      to_add = Enum.reduce(Map.get(member_spec, "add", []),
-                           %{found: [], not_found: []}, &lookup_or_fail/2)
-      to_remove = Enum.reduce(Map.get(member_spec, "remove", []),
-                           %{found: [], not_found: []}, &lookup_or_fail/2)
+      to_add = Enum.reduce(Map.get(member_spec, "add", []), %{found: [], not_found: []},
+                           &(lookup_or_fail(type, &1, &2)))
+      to_remove = Enum.reduce(Map.get(member_spec, "remove", []), %{found: [], not_found: []},
+                              &(lookup_or_fail(type, &1, &2)))
+
       case to_add.not_found ++ to_remove.not_found do
         [] ->
-          Queries.RelayGroup.add_relays!(group.id, to_add.found)
-          removed = Queries.RelayGroup.remove_relays(group.id, to_remove.found)
+          Queries.RelayGroup.add!(type, group.id, to_add.found)
+          removed = Queries.RelayGroup.remove(type, group.id, to_remove.found)
           if removed != length(to_remove.found) do
-            Repo.rollback(:removing_relays_failed)
+            Repo.rollback(:remove_failed)
           else
-            Repo.preload(group, :relays)
+            group
+            |> Repo.preload(:bundles)
+            |> Repo.preload(:relays)
           end
         ids ->
           Repo.rollback({:unknown_relays, ids})
```

Comment:
Is `:unknown_relays` still accurate given the controller's expanded functionality?